### PR TITLE
@ #65 | should upgrade ubuntu image to 18.04

### PR DESCRIPTION
**Pull request by hieptranquoc**
connect #65 
please check this PR!

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,71 +1,71 @@
 services:
-- docker
+  - docker
```
Comment (hoatle) [inline .travis.yml]: don't try to reformat to minimize the unnecessary changes for easier, faster reviewing.

File: ubuntu/base/Dockerfile [hunk 2]
```diff
@@ -10,19 +10,27 @@ ARG CI_REGISTRY_IMAGE
 ARG CI_BUILD_TIME
 
 ENV CI_BUILD_ID=$CI_BUILD_ID CI_BUILD_REF=$CI_BUILD_REF CI_REGISTRY_IMAGE=$CI_REGISTRY_IMAGE \
-    CI_BUILD_TIME=$CI_BUILD_TIME
+  CI_BUILD_TIME=$CI_BUILD_TIME
```
Comment (hoatle) [inline ubuntu/base/Dockerfile]: avoid reformating changes like this.

File: ubuntu/base/install.sh [hunk 3]
```diff
@@ -1,61 +1,42 @@
...
-    wget \
-    curl \
-    python \
-    lsb-release \
-    apt-transport-https \
-    ca-certificates \
-    software-properties-common \
-&& apt-get clean && rm -rf /var/lib/apt/lists/* \
-; \
+  wget \
```
Comment (hoatle) [inline ubuntu/base/install.sh]: keep the existing alignment format, don't try to change when it's unnecessary.

**Comment by hieptranquoc** [comment]
Applied all comments! Please check again!

**Comment by hoatle** [comment]
@hieptranquoc make sure we can use ubuntu versions (14.04, 16.04, 18.04) to install (any) recent docker and docker-compose versions, so please add more builds and images to reflect that for the travis-ci to build to confirm.

**Comment by hoatle** [comment]
ok with docker:18-dind image

```
macbook-pro:base hoatle$ docker run --privileged --name docker -d -v $(pwd):$(pwd) -w $(pwd) docker:18-dind
Unable to find image 'docker:18-dind' locally
18-dind: Pulling from library/docker
9d48c3bd43c5: Already exists 
7f94eaf8af20: Already exists 
9fe9984849c1: Already exists 
3091f1b4f1aa: Pull complete 
6ef266ac0949: Pull complete 
b2c2c13f4c08: Pull complete 
f354b3ae6d74: Pull complete 
8f4a6170836f: Pull complete 
853fedec02a1: Pull complete 
a57a377d7e5d: Pull complete 
ac4bc61da695: Pull complete 
Digest: sha256:86df3c3573065f2c6f24cd925fd5bc3a0aff899bdf664ff4d2e3ebab26d96bed
Status: Downloaded newer image for docker:18-dind
e77c5bf60df30bd2a7b0068e9a03878e2c6384bb0086ead8a07c4b312cbaf8d0
macbook-pro:base hoatle$ docker run --rm -it --link docker -v $(pwd):$(pwd) -w $(pwd) teracy/ubuntu:16.04-dind-latest bash
root@720370168fde:/Users/hoatle/teracy-dev/workspace/docker-files/ubuntu/base# docker version
Client: Docker Engine - Community
 Version:           19.03.3
 API version:       1.39 (downgraded from 1.40)
 Go version:        go1.12.10
 Git commit:        a872fc2
 Built:             Tue Oct  8 00:59:54 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.9
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.11.13
  Git commit:       039a7df9ba
  Built:            Wed Sep  4 16:55:50 2019
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Comment by hoatle** [comment]
NOT ok with docker:19-dind image:

```
macbook-pro:base hoatle$ docker run --privileged --name docker -d -v $(pwd):$(pwd) -w $(pwd) docker:19-dind
627ef99d7d1ca44dadefd78135ae63254de2b4ae55dd6492adb307f18e03e8cc
macbook-pro:base hoatle$ docker run --rm -it --link docker -v $(pwd):$(pwd) -w $(pwd) teracy/ubuntu:16.04-dind-latest bash
root@fc3eacf32c45:/Users/hoatle/teracy-dev/workspace/docker-files/ubuntu/base# docker version
Client: Docker Engine - Community
 Version:           19.03.3
 API version:       1.40
 Go version:        go1.12.10
 Git commit:        a872fc2
 Built:             Tue Oct  8 00:59:54 2019
 OS/Arch:           linux/amd64
 Experimental:      false
Cannot connect to the Docker daemon at tcp://docker:2375. Is the docker daemon running?
```

**Comment by hoatle** [comment]
for docker:19-dind, set `-e DOCKER_TLS_CERTDIR=""`:

```
macbook-pro:base hoatle$ docker run --privileged --name docker -d -v $(pwd):$(pwd) -w $(pwd) -e DOCKER_TLS_CERTDIR="" docker:19-dind
52b0e323e4ae606bfe7d4a5abf9ad98e275858e349c5986ee0e13c7bf4d21cf3
macbook-pro:base hoatle$ docker run --rm -it --link docker -v $(pwd):$(pwd) -w $(pwd) teracy/ubuntu:16.04-dind-latest bash
root@8f04217dbeb1:/Users/hoatle/teracy-dev/workspace/docker-files/ubuntu/base# docker version
Client: Docker Engine - Community
 Version:           19.03.3
 API version:       1.40
 Go version:        go1.12.10
 Git commit:        a872fc2
 Built:             Tue Oct  8 00:59:54 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.3
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.10
  Git commit:       a872fc2f86
  Built:            Tue Oct  8 01:01:20 2019
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.10
  GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
 runc:
  Version:          1.0.0-rc8+dev
  GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
root@8f04217dbeb1:/Users/hoatle/teracy-dev/workspace/docker-files/ubuntu/base# 
```

related:
- https://gitlab.com/gitlab-org/gitlab-runner/issues/4501
- https://github.com/docker-library/docker/issues/170
